app/models/project/scopes/visible_with_activated_time_activity.rb

```diff
@@ -0,0 +1,52 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2018 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2017 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module Project::Scopes
+  class VisibleWithActivatedTimeActivity
```

Comment:
I played around with the module_function approach now but ended up not liking it because of the limitation to declare methods private.

I intend to spec every public methods in the scopes heavily as they are one of the corner stones of the application with them being e.g. used for validations. When more methods are public I would have to spec more methods (with the specs overlapping).

As for the reuse: In case we find ourselves needing only part of the scope, it should be fairly simple to extract that part into its own scope and use it then.

Also, I do not see the Scopes as something that should ever be included. Where would the benefit be? They are not intended to alter the behaviour of an including class.

But I am aware that this is more personal preference than hard facts, so I would have no problems with scopes written differently.

Reply:
As you said it's a bit of a matter of taste. In that case let's just leave it at that. It doesn't really make that much of a difference.